Makefile

```diff
@@ -57,7 +57,13 @@ build:
 	@echo "$(GREEN)Pulling Docker image...$(RESET)"
 	@docker pull $(DOCKER_IMAGE)
 	@echo "$(GREEN)Installing Python dependencies...$(RESET)"
-	@curl -sSL https://install.python-poetry.org | python3 -
+	@if command -v poetry > /dev/null; then \
+		echo "$(BLUE)Poetry is already installed.$(RESET)"; \
+	else \
+		echo "$(YELLOW)Poetry is not installed. You can install poetry by running the following command, then adding Poetry to your PATH:"; \
+		echo "$(YELLOW)curl -sSL https://install.python-poetry.org | python3 -$(RESET)"; \
```

Comment:
I would recommend to put the installation link to remove any confusion and handle what poetry doc support 👌 what do you think about that?

Reply:
Also moving this check before pulling the docker image would give a seamless installation so the users will not wait till the end of the pulling then having an error. It's not a big think HHHH

Reply:
Hey @dorbanianas , what do you mean by "put the installation link"?

Reply:
I mean referring the user to poetry official installation website: https://python-poetry.org/docs/#installing-with-pipx

Reply:
So this would help to have a global solution and giving more details

Reply:
Sure, added.